openshift/openshift/src/main/java/org/arquillian/cube/openshift/impl/client/PortForwardServerConnection.java

```diff
@@ -0,0 +1,354 @@
+    @Override
+    public HttpServerExchange sendOutOfBandResponse(HttpServerExchange exchange) {
+        throw new UnsupportedOperationException("PortForward connection does not support HTTP!");
+    }
+
+    @Override
+    public void terminateRequestChannel(HttpServerExchange exchange) {
+        throw new UnsupportedOperationException("PortForward connection does not support HTTP!");
+    }
+
+    @Override
+    public SSLSessionInfo getSslSessionInfo() {
+        // We're not supporting SSL
+        return null;
+    }
+
+    @Override
+    public void setSslSessionInfo(SSLSessionInfo sessionInfo) {
+        throw new UnsupportedOperationException("PortForward connection does not support SSL!");
+    }
+
+    @Override
+    protected StreamConnection upgradeChannel() {
+        throw UndertowMessages.MESSAGES.upgradeNotSupported();
+    }
+
+    @Override
+    protected StreamSinkConduit getSinkConduit(HttpServerExchange exchange, StreamSinkConduit conduit) {
+        return conduit;
+    }
+
+    @Override
+    protected boolean isUpgradeSupported() {
+        return false;
+    }
+
+    @Override
+    protected void exchangeComplete(HttpServerExchange exchange) {
+        // We're not supporting HTTP so nothing to do here
+    }
+
+    @Override
+    public String getTransportProtocol() {
+        return "raw";
+    }
+
+    @Override
+    protected boolean isConnectSupported() {
+        return false;
+    }
+
+    @Override
+    public boolean isContinueResponseSupported() {
+        return false;
+    }
+
+    @Override
+    protected void setConnectListener(HttpUpgradeListener connectListener) {
+    }
+
+    public void startForwarding(final ClientConnection clientConnection, final String urlPath, final int targetPort, final int requestId) throws IOException {
+        try {
+            //resetChannel();
+//            pipe = getIoThread().createFullDuplexPipeConnection(clientConnection.getIoThread());
+//            // hook up the left side
+//            ChannelUtils.initiateTransfer(Long.MAX_VALUE, getChannel().getSourceChannel(), pipe.getLeftSide().getSinkChannel(), bufferPool);
+//            ChannelUtils.initiateTransfer(Long.MAX_VALUE, pipe.getLeftSide().getSourceChannel(), getChannel().getSinkChannel(), bufferPool);
+
+            // initiate the streams
+            openErrorStream(clientConnection, urlPath, targetPort, requestId);
+            openDataStream(clientConnection, urlPath, targetPort, requestId);
+            try {
+                /*
+                 * wait for the request to complete. this will trigger when the
+                 * client is done and the request stream closes.
+                 */
+                requestComplete.await();
+                /* wait for the response on the error stream. */
+                errorComplete.await();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
This printStackTrace will hide a possible exception, is this intentionally done or needs some refactor this part of code.